src/Talis/Persona/Client/Base.php

```diff
@@ -55,40 +57,50 @@
     private $defaultTtl;
 
     /**
+     * @var string
+     */
+    private $appUserAgent;
+
+    /**
      * Constructor
      *
+     * @param string $appUserAgent Consuming application user agent string @since 2.0.0
+     *      examples: rl/5.2 (php/5.3), rl, rl/5, rl/5.2(php/5.3; linux/2.5)
      * @param array $config An array of options with the following keys: <pre>
      *      persona_host: (string) the persona host you'll be making requests to (e.g. 'http://localhost')
      *      persona_oauth_route: (string) the token api route to query ( e.g: '/oauth/tokens')
      *      cacheBackend: (Doctrine\Common\Cache\CacheProvider) optional cache storage (defaults to Filesystem)
      *      cacheKeyPrefix: (string) optional prefix to append to the cache keys
      *      cacheDefaultTTL: (integer) optional cache TTL value
-     * @param @deprecated $logger optional logger instance
-     * @param @deprecated $cacheBackend optional cache backend see
-     *  https://github.com/doctrine/cache/tree/master/lib/Doctrine/Common/Cache
-     *  https://doctrine-orm.readthedocs.org/projects/doctrine-orm/en/latest/reference/caching.html
      * @throws \InvalidArgumentException if any of the required config parameters are missing
-     * 
+     * @throws \InvalidArgumentException if the user agent format is invalid
      */
     public function __construct(
-        array $config,
-        \Psr\Log\LoggerInterface $logger = null,
-        CacheProvider $cacheBackend = null
+        $appUserAgent,
```

Comment:
Is this what you want to be doing here? Changing the signature so that UA is now the first param and config the 2nd? You've just deprecated 2 args, would it make more sense to add the UA to config?